EFCore.NamingConventions/Internal/NameRewritingConvention.cs

```diff
@@ -57,10 +57,10 @@ public NameRewritingConvention(INameRewriter nameRewriter)
     {
         var entityType = entityTypeBuilder.Metadata;
 
-        if (newBaseType is null)
+        if (newBaseType is null || entityType.GetMappingStrategy() == RelationalAnnotationNames.TpcMappingStrategy)
         {
             // The entity is getting removed from a hierarchy. Set the (rewritten) TableName.
-            if (entityType.GetTableName() is { } tableName)
+            if (entityType.GetTableName() is { } tableName && !entityType.ClrType.IsAbstract)
```

Comment:
this obviously breaks that all abstract tables are not rewritten and would stay in a different case, when UseSnakeCaseNamingConvention is applied the tablename would stay in normal case, which would break a lot of existing code and tables.
related to #184